packages/widget/src/components/address-input/address-input.ts

```diff
@@ -0,0 +1,83 @@
+import { LitElement, html } from 'lit';
+import type { HTMLTemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { Network } from '@buildwithsygma/sygma-sdk-core';
+import { when } from 'lit/directives/when.js';
+import { validateAddress } from '../../utils';
+import { styles } from './styles';
+
+@customElement('sygma-address-input')
+export class AddressInput extends LitElement {
+  static styles = styles;
+  @property({
+    type: String
+  })
+  address: string = '';
+
+  @property({ attribute: false })
+  onAddressChange: (address: string) => void = () => {};
+
+  @property({
+    type: String
+  })
+  network: Network = Network.EVM;
+
+  @state()
+  errorMessage?: string;
```

Comment:
i suggest making this non-optional (since we don't need a three-state, undefined/''/'sum ting wong') and if it's an empty string (falsy value), all good, if no, error!

Reply:
even better, to be more explicit, i think it'd be better to put 
```suggestion
  errorMessage: string | null = null;
```